### PR TITLE
Convert to GHA, fix tests, fix build

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -5,25 +5,26 @@ version: "3.1"
...
       workspace_url: https://ci.kbase.us/services/ws
+      catalog_url: https://ci.kbase.us/services/catalog
+      notifier_url: https://ci.kbase.us/services/feeds
       workspace_admin_token: faketoken_replace
+      notifier_token: faketoken_replace
       field_validator_configs: |
          field-foo-validator=us.kbase.groups.fieldvalidators.SimpleFieldValidatorFactory
          field-foo-is-numbered=true
       # add more configurations here
     command:
```
Comment (Xiangs18) [inline docker-compose.yml:24]: If my understanding is correct, 
a line starts with "-" is an arg, and then followed by its val? For instance, "-timeout", "120s"
What are "java" and "-Djetty.home=/usr/local/jetty" then?

File: docker-compose.yml [hunk 2]
```diff
@@ -5,25 +5,26 @@ version: "3.1"
...
       - "-multiline"
       - "-wait"
-      - "tcp://ci-mongo:27017"
-      - "-wait"
-      - "tcp://mongoinit:8080"
+      - "tcp://mongo:27017"
       - "-timeout"
       - "120s"
       - "-template"
```
Comment (Xiangs18) [inline docker-compose.yml:30]: If my understanding is correct,
a line starts with "-" is an arg, and then followed by its val? For instance, "-timeout", "120s"
What are "java" and "-Djetty.home=/usr/local/jetty" then?

Comment (MrCreosote) [reply docker-compose.yml:30]: > "-" is an arg, and then followed by its val?

Not always, one example is `-multiline`

The first set of arguments are for `dockerize`. Then there's the command that `dockerize` will run, `java`, and its arguments.

Comment (MrCreosote) [reply docker-compose.yml:30]: everything after `java` are the java args

Comment (MrCreosote) [reply docker-compose.yml:30]: the arguments are for `dockerize`, not `docker-compose`

Comment (Xiangs18) [reply docker-compose.yml:30]: I see. May I know which line it pull `dockerize` image? Cannot find it anywhere.

Comment (MrCreosote) [reply docker-compose.yml:30]: It doesn't, dockerize is built into the base image

Comment (Xiangs18) [reply docker-compose.yml:30]: I found it. is here:
```
|3 BRANCH=master BUILD_DATE=2018-11-08T20:13:52Z VCS_REF=bb0c445 /bin/sh -c mkdir -p /var/lib/apt/lists/partial && 	apt-get update -y &&     apt-get install -y ca-certificates tomcat8-user libservlet3.1-java &&     update-ca-certificates &&     useradd -c "KBase user" -rd /kb/deployment/ -u 998 -s /bin/bash kbase &&     mkdir -p /kb/deployment/bin &&     mkdir -p /kb/deployment/jettybase/logs/ &&     touch /kb/deployment/jettybase/logs/request.log &&     chown -R kbase /kb/deployment &&     cd /kb/deployment/bin &&     wget -N https://github.com/kbase/dockerize/raw/master/dockerize-linux-amd64-v0.6.1.tar.gz && 	tar xvzf dockerize-linux-amd64-v0.6.1.tar.gz &&     rm dockerize-linux-amd64-v0.6.1.tar.gz
```

**Review by Xiangs18 — APPROVED** [review]
LGTM!

**Pull request merged**